docs/source/download.rst

```diff
@@ -165,6 +127,88 @@ type::
 
 You should now be able to run ``firedrake-update``.
 
+System requirements
+-------------------
+
+Firedrake requires Python 3.6 or later. The installation script is
+tested on Ubuntu and MacOS X. On Ubuntu 18.04 or later, the system
+installed Python 3 is supported and tested. On MacOS, the homebrew_
+installed Python 3 is supported and tested::
+
+  brew install python3
+
+Installation is likely to work well on other Linux platforms, although
+the script may stop to ask you to install some dependency packages.
+Installation on other Unix platforms may work but is untested. On Linux
+systems that do not use the Debian package management system, it will be
+necessary to pass the `--no-package-manager` option to the install
+script. In this case, it is the user's responsibilty to ensure that
+they have the system dependencies:
+
+* A C and C++ compiler (for example gcc/g++ or clang), GNU make
+* A Fortran compiler (for PETSc)
+* Blas and Lapack
+* Git, Mercurial
+* Python version >=3.6
+* The Python headers
+* autoconf, automake, libtool
+* CMake
+* zlib
+* flex, bison
+
+Firedrake has been successfully installed on Windows 10 using the
+Windows Subsystem for Linux. There are more detailed
+`instructions here <https://github.com/firedrakeproject/firedrake/wiki/Installing-on-Windows-Subsystem-for-Linux>`_.
+Installation on previous versions of Windows is unlikely to work.
+
+
+System anti-requirements
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+We strive to make Firedrake work on as many platforms as we can. Some
+tools, however, make this challenging or impossible for end users.
+
+**Anaconda.** The Anaconda Python distribution and package manager are
+often recommended in introductory data science courses because it does
+effectively handle many aggravating problems of dependency management.
+Unfortunately, Anaconda does a poor job of isolating itself from the
+rest of your system and assumes that it will be both the only Python
+installation and the only supplier of any dependent packages. Anaconda
+will install compilers and MPI compiler wrappers and put its compilers
+right at the top of your ``PATH``. This is a problem because Firedrake
+needs to build and use its own MPI. (We keep our MPI isolated from the
+rest of your system through virtual environments.) When installed on a
+platform with Anaconda, Firedrake can accidentally try to link to the
+incompatible Anaconda installation of MPI.
+
+There are three ways to work around this problem.
+
+1. Remove Anaconda entirely.
+2. Modify your ``PATH`` environment variable to remove any traces of
+   Anaconda, then install Firedrake. If you need Anaconda later, you
+   can re-enable it with a shell script that will add those directories
+   back onto your path.
+3. Use a tool like pyenv_, which makes it possible to switch between
+   different versions of Python on a per-terminal session or per-
+   directory basis.
```

Comment:
An FYI - pyenv broke my installation terribly on my mac when I tried to update, presumably because I didn't use pyenv's virtual env tool. I spent quite a long time working out what went wrong. Not sure if it's a great idea to suggest using it over, say, one of the docker images.